procedures/automation-android-setup.md

```diff
@@ -0,0 +1,200 @@
+# How to set up automation for nightly Android product
+
+These instructions define which resources are needed for setting up nightly Android product release automation, as well as the steps to prepare the automation itself. Additionally, this guide include steps for running a "staging" version of the automation so that it can be tested in the developer's own repository.
+
+Throughout this document, wherever the term `$product` is used, substitute your product's name in (replacing spaces with hyphens), e.g.: `reference-browser` or `fenix`.
+
+Note: we don't need to explicitly define scopes in Taskcluster. We just define them in the task, hook's role, and each scriptworker script. Taskcluster will implicitly detect scopes, but it doesn't use them for anything other than generic equality/wildcard verification.
+
+1. Request dep and release signing keys (example bug for [`reference-browser`](https://bugzilla.mozilla.org/show_bug.cgi?id=1508761)). Confirm with the app's team what the "signature algorith" and "digest algorithm" should be, and include that information in the ticket.
+2. Clone the product's repository
+    * Note that, while doing "staging" runs, you'll have to use your `master` branch in your fork. Hook cloning/CoT/etc haven't been tested with a non-default branch
+3. Add `.taskcluster.yml` in the root of the repository if it doesn't exist, and add a task for cron jobs. ([`fenix` example, go to `tasks_for == "cron"`](https://github.com/mozilla-mobile/fenix/blob/master/.taskcluster.yml))
+    1. Update the `event.repository.clone_url` checks to use the correct repo URL
+    2. Update scopes to be relevant for your product
+    3. Update the apks in `in.payload.command` so that the paths match your product
+    4. Update `in.payload.image` so that a compatible docker image is used
+    5. Update `in.metadata` for your product
+4. In `automation/taskcluster`, add `decision_task_nightly.py`, `schedule_nightly_graph.py` and `lib/tasks.py` that will perform most of the automation wiring. ([`fenix` examples](https://github.com/mozilla-mobile/fenix/blob/master/automation/taskcluster/))
```

Comment:
In the future we should avoid to create `decision_task_nightly.py`. Instead: 
* either we modify `decision.py` (example: https://github.com/mozilla-mobile/android-components/pull/1842) 
* or taskgraph will avoid the need of having decision_task_nightly.py

Reply:
I'd like to wait for `taskgraph` for this migration, if y'alls are :ok_hand: with that.

Reply:
Sounds good 👍